File: my-folder/0128-longest-consecutive-sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums)
        nums = list(numSet)
        res = 0
        # we need to figure out if the number we are dealing with, is it the starting number of a sequence
        for i in range(len(nums)):
            if nums[i] - 1 not in numSet:   # that means the current number, is the starting of sequence
                counter = 0
                currNum = nums[i]
                while currNum in numSet:
                    counter += 1
                    currNum += 1
                res = max(res, counter)
        return res
                

File: my-folder/0128-longest-consecutive-sequence/solution.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        HashSet<int> numSet = new (nums);
        int longest = 0;
        foreach(int ele in numSet) {
            if (!numSet.Contains(ele - 1)) {
                int length = 1;

                while (numSet.Contains(ele + length)) {
                    length += 1;
                }
                longest = Math.Max(longest, length);
            }
        }
        return longest;
    }
}
